scipy/sparse/data.py

```diff
@@ -181,6 +193,63 @@ def _min_or_max(self, axis, out, min_or_max):
         else:
             raise ValueError("axis out of range")
 
+    def _arg_min_or_max(self, axis, out, op, compare):
+        if out is not None:
+            raise ValueError("Sparse matrices do not support "
+                             "an 'out' parameter.")
+
+        if 0 in self.shape:
+            raise ValueError("Can't apply the operation to an empty matrix.")
+
+        validateaxis(axis)
+
+        zero = self.dtype.type(0)
+
+        if axis is None:
+            if self.nnz == 0:
+                return 0
+            else:
+                mat = self.tocoo()
+                mat.sum_duplicates()
+                am = op(mat.data)
+                m = mat.data[am]
+
+                if compare(m, zero):
+                    return mat.row[am] * mat.shape[1] + mat.col[am]
+                else:
+                    size = np.product(mat.shape)
+                    if size == mat.nnz:
+                        return am
+                    else:
+                        ind = mat.row * mat.shape[1] + mat.col
+                        return _find_missing_index(ind, size)
+        else:
+            if axis < 0:
+                axis += 2
+
+            mat = self.tocsc() if axis == 0 else self.tocsr()
+            mat.sum_duplicates()
+
+            line_size = mat.shape[axis]
+            ret = np.empty(mat.shape[1 - axis], dtype=int)
+
+            for i in range(ret.shape[0]):
```

Comment:
I don't think we can avoid the loop entirely, but we can at least vectorize the first condition:
```python
nz_lines, = np.diff(mat.indptr) > 0
for i in nz_lines:
    p, q = mat.indptr[i:i+2]
    data = mat.data[p:q]
    # etc...
```